src/stack/value.rs

```diff
@@ -0,0 +1,28 @@
+use std::sync::Arc;
+
+use {Any, Interned, Node, Push, Pop, Stack};
+
+pub enum Value {
+    Array(Vec<Value>),
+    Bool(bool),
+    Custom(Box<Any>),
+    Float(f64),
+    Int(i64),
+    Interned(Interned),
+    Node(Arc<Node>),
+    Plain,
+    String(String),
+    Tuple(Vec<Value>),
+    Uint(u64),
+    Void,
+}
+
+impl Push for Value {
```

Comment:
This is the problematic part currently, because `Value` can't be pushed like everything else. I need to know which variant it is, but I can't get a reference to the value because `push_value` gets called before `push_tags` so the value is consumed already. I cannot make them panic and implement the whole `push` method, because e.g. tuples push the tags separately from the values, so we can first check all the types match and restore the tags in case they don't. If I would mix values and tags that would be a lot more complicated..

Reply:
Okay I just decided how I do it: In case there's an error, I don't restore the removed tags and values but simply remove the rest. I think that's what makes most sense. If you pass arguments to a function and the arguments don't match, there's no reason to leave them on the stack.